### ISSUE TITLE
Automatically prepend `async` when typing `.await` in a function

### ISSUE THREAD

**Issue by nilsmartel** [rust-lang/rust-analyzer#10111]
The title pretty much states, what I think would be a great use case for rust analyzer.

Here's the conversation that got this idea rolling: https://twitter.com/minigamedev/status/1433075817110810626

here's the expected behaviour:
when a use types .await, like this:
```rust
fn abc() {
   x.await
}
```

the function should automatically be rewritten to read:

```rust
async fn abc() {
    x.await
}
```

If I understood it correctly.

**Comment by bjorn3** [comment]
I don't know if it is possible to add `async` if you are done typing `await` normally. What would be possible is having the `await` keyword completion add it as part of completing `await`, which would require accepting the completion. This would be similar to trait auto-importing which is triggered by a completion too.

**Comment by matklad** [comment]
I don't think we should auto-fix code on completion (auto import being an exception). We should emit a diagnostic for `.await` in non-async function to turn the function async (and update all the call-sites to `.await` it).

As a meta strategy, IDE should follow the "unix philosophy" and provide composable building blocks. :bulb: is the primary operator for such composition.

**Comment by ChayimFriedman2** [comment]
We now have a diagnostic but without a fix.
